backend/ecommerce/models.py

```diff
@@ -1,7 +1,36 @@
 from django.db import models
+from django.contrib.auth.models import AbstractBaseUser, PermissionsMixin
+from django.core.validators import MinValueValidator, MaxValueValidator
+from django.utils.translation import ugettext_lazy as _
 
-# Create your models here.
+from .managers import UserManager
 
 
-class Hii(models.Model):
-    name = models.IntegerField()
+class User(AbstractBaseUser, PermissionsMixin):
+    name = models.CharField(max_length=255)
+    email = models.EmailField(max_length=255, unique=True)
+    is_staff = models.BooleanField(
+        default=False, help_text=_("Designates if a user can log into this admin site")
+    )
+    phone = models.IntegerField(
+        validators=[MinValueValidator(1000000000), MaxValueValidator(9999999999)]
+    )
+
+    USERNAME_FIELD = "email"
+
+    REQUIRED_FIELDS = ["name", "phone"]
```

Comment:
Isn't everything required by default?

Reply:
These are the fields which will be asked while creating the super user.